core/field.js

```diff
@@ -52,6 +52,52 @@ Blockly.Field = function(text, opt_validator) {
   this.setValidator(opt_validator);
 };
 
+/**
+ * The set of all registered fields, keyed by field type as used in the JSON
+ * definition of a block.
+ * @private
+ */
+Blockly.Field.ALL_ = {};
+
+/**
+ * Registers a field type. May also override an existing field type.
+ * Blockly.Field.fromJson uses this registry to find the appropriate field.
+ * @param {!string} type The field type name as used in the JSON definition.
+ * @param {!{fromJson: Function}} fieldClass The field class containing a
+ *                                 fromJson function that can construct an
+ *                                 instance of the field.
+ * @throws {Error} if the type name is empty, or the fieldClass is not an
+ *                 object containing a fromJson function.
+ */
+Blockly.Field.register = function(type, fieldClass) {
+  if (!goog.isString(type) || goog.string.isEmptyOrWhitespace(type)) {
+    throw new Error('Error: Invalid field type "' + type + '"');
```

Comment:
Nit: no need for the leading `'Error:` text, since the browser will prepend something like that anyway.  Here and below.

Reply:
The errors in Blockly.Extensions probably need updating too then